src/github.ts

```diff
@@ -41,26 +56,44 @@ export function getTempDir(): string {
 
 function getPullRequestContext(
   context: Context
-): Pick<GitHubContext, "prNumber" | "sha"> {
-  const number = context.issue.number;
+): PullRequestInfo {
+  const prNumber = context.issue.number;
   const sha = context.payload.pull_request?.head.sha;
-  return { prNumber: number, sha };
+
+  return { prNumber, sha };
 }
 
 function getCheckRunContext(
   context: Context
-): Pick<GitHubContext, "prNumber" | "sha"> {
+): PullRequestInfo {
   const actionEvent = context.payload.check_run;
-  const number = actionEvent.pull_requests?.[0]?.number;
+  const prNumber = actionEvent.pull_requests?.[0]?.number;
   const sha = actionEvent.head_sha;
-  return { prNumber: number, sha };
+
+  return { prNumber, sha };
+}
+
+/**
+ * @param context The GitHub context object containing information about the event.
+ * @returns A Promise that resolves to a PullRequestInfo object with the SHA of the commit.
+ */
+async function getWorkflowDispatchContext(context: Context): Promise<PullRequestInfo> {
+  const branchName: string = context.ref.substring('refs/heads/'.length);
+  const defaultBranch: string = context.payload.repository?.default_branch;
+
+  let prNumber;
+  if (branchName !== defaultBranch) {
```

Comment:
I'm confused with this line, so if the dispatch is executed in a branch different that the main one, we need to pass the pr-number value, but if they are the same we dont need to pass it? I thought you told me that you coudl get the prNumber from the paylod of the action, when it was from branch